File: backend/src/main/java/com/yangshengzhou/lucky_sms/pojo/Course.java
package com.yangshengzhou.lucky_sms.pojo;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;

/**
 * 课程实体类
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("courses")
public class Course extends BaseEntity {
    
    @TableId(type = IdType.AUTO)
    private Integer courseId;
    
    @TableField("course_code")
    private String courseCode;
    
    @TableField("course_name")
    private String courseName;
    
    @TableField("course_type_id")
    private Integer courseTypeId;
    
    private BigDecimal credit;
    
    @TableField("total_hours")
    private Integer totalHours;
    
    @TableField("theory_hours")
    private Integer theoryHours;
    
    @TableField("practice_hours")
    private Integer practiceHours;
    
    @TableField("department_id")
    private Integer departmentId;
    
    @TableField("major_id")
    private Integer majorId;
    
    private String description;
    private String syllabus;
    private String textbook;
    private String prerequisites;
    
    @TableField("is_compulsory")
    private Boolean isCompulsory;
    
    @TableField("status_id")
    private Integer statusId;
    
    // 关联查询字段
    @TableField(exist = false)
    private Department department;
    
    @TableField(exist = false)
    private Major major;
    
    @TableField(exist = false)
    private CourseType courseType;
    
    @TableField(exist = false)
    private StatusType status;
}

File: backend-go/models/course.go
package models

import (
	"math/big"
)

type Course struct {
	BaseEntity
	CourseID      int         `json:"courseId" gorm:"column:course_id;primaryKey;autoIncrement"`
	CourseCode    string      `json:"courseCode" gorm:"column:course_code;size:20;uniqueIndex"`
	CourseName    string      `json:"courseName" gorm:"column:course_name;size:100"`
	CourseTypeID  int         `json:"courseTypeId" gorm:"column:course_type_id;index"`
	Credit        *big.Float  `json:"credit" gorm:"column:credit;type:decimal(3,1)"`
	TotalHours    int         `json:"totalHours" gorm:"column:total_hours"`
	TheoryHours   int         `json:"theoryHours" gorm:"column:theory_hours"`
	PracticeHours int         `json:"practiceHours" gorm:"column:practice_hours"`
	DepartmentID  int         `json:"departmentId" gorm:"column:department_id;index"`
	MajorID       int         `json:"majorId" gorm:"column:major_id;index"`
	Description   string      `json:"description" gorm:"column:description;type:text"`
	Syllabus      string      `json:"syllabus" gorm:"column:syllabus;type:text"`
	Textbook      string      `json:"textbook" gorm:"column:textbook;size:255"`
	Prerequisites string      `json:"prerequisites" gorm:"column:prerequisites;type:text"`
	IsCompulsory  bool        `json:"isCompulsory" gorm:"column:is_compulsory;default:true"`
	StatusID      int         `json:"statusId" gorm:"column:status_id"`
	
	Department *Department `json:"department,omitempty" gorm:"-"`
	Major      *Major      `json:"major,omitempty" gorm:"-"`
	CourseType *CourseType `json:"courseType,omitempty" gorm:"-"`
	Status     *StatusType `json:"status,omitempty" gorm:"-"`
}

func (Course) TableName() string {
	return "courses"
}
